### ISSUE TITLE
Trouble running the extractor

### ISSUE THREAD

**Issue by tumbleshack** [rust-corpus/qrates#78]
Hi there! I'm an admirer of your work and your associated paper; I enjoyed learning about unsafe usage.

We (@seanmichwalsh and I) are attempting to run your experiment on a set of crates from crates.io, but we can't get the extractor to run as part of `cargo run --release -- compile` when following [these instructions](https://rust-corpus.github.io/qrates/extractor_crates_io.html). It compiles the crates under analysis successfully, but does not run the extractor on them.

Setting the environment variables as shown [on this page](https://rust-corpus.github.io/qrates/extractor_private.html) does indeed run the extractor, but causes the following problem, which stops execution: `error: output of --print=file-names has changed in the compiler, cannot parse`. Full log [on this pastebin](https://pastebin.com/kkHW4gBT).

How can we run the extractor properly? Thanks!

**Comment by vakaras** [comment]
I guess you are running on Mac OS, right? I have never tested on a system other than Ubuntu, so I have no idea what could be wrong.

This error hints at that your Rust installation could be broken, but I am not sure:
```
error: output of --print=file-names has changed in the compiler, cannot parse
```

My suggestion would be to get an Ubuntu VM and try inside it. As far as I know, Docker on Mac OS is running inside a VM anyway, so this approach should not have any negative performance impact.

**Comment by tumbleshack** [comment]
Yes, I am on Mac. @seanmichwalsh does repro on Ubuntu 20.04, however

Inside my Ubuntu VM, running `cargo run --release -- compile` without setting environment variables does not run the extractor. I'll try setting the env vars and seeing how it goes

**Comment by tumbleshack** [comment]
We solved this problem on Ubuntu 20.04. We changed three things:
1. We removed and did a fresh installation of Rust
2. Reverted to commit [`b20cf9` ](https://github.com/rust-corpus/qrates/tree/b20cf99fa640be86cee54649ff8ed521d4666af3)
3. Instead of `sudo env "PATH=$PATH" $(which cargo) run --release -- compile`, we ran `sudo --preserve-env $(which cargo) run --release -- compile`

**Comment by vakaras** [comment]
Thanks! I will investigate what is going on here and why CI does not catch this once I have a free moment.

**Comment by tumbleshack** [comment]
Awesome. It is worth noting then, that the test_extractor.py passed successfully even when the “cannot parse” problem was encountered.

**Comment by vakaras** [comment]
I tried to reproduce the issue by following the instructions on a freshly checked out repo. However, the current master seems to work for me without issues. In all cases, I get the `*.bincode` files as expected.

**Comment by vakaras** [comment]
Could you please check whether the latest master branch works for you? Maybe it was an issue with the version of the Rust compiler?
